projects/js-packages/initial-state/src/utils.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Get the initial state from the window object.
+ *
+ * @returns {import('./types').JetpackInitialState} The initial state.
+ */
+export function getInitialState() {
+	return window.JETPACK_INITIAL_STATE;
+}
+
+/**
+ * Get the site data from the initial state.
+ *
+ * @returns {import('./types').SiteData} The site data.
+ */
+export function getSiteData() {
+	return getInitialState().site;
+}
+
+/**
+ * Get the admin URL from the initial state.
+ *
+ * @param {string} [path] - The path to append to the admin URL. e.g. `admin.php?page=jetpack`.
+ *
+ * @returns {string} The admin URL.
+ */
+export function getAdminUrl( path = '' ) {
+	return `${ getInitialState().site.admin_url }${ path }`;
```

Comment:
I couldn't get to actually test this because it's not implemented by other packages, but won't this cause URLs to miss a forward slash after the admin URL? like `https://yoursite.comadmin.php?page...`?

Reply:
You can see its use just below there in `getMyJetpackUrl`

Reply:
> [`admin_url`](https://developer.wordpress.org/reference/functions/admin_url/) is always with a trailing slash. What we might want to do here is `ltrim()` as done by [`get_admin_url`](https://developer.wordpress.org/reference/functions/get_admin_url/). Do we need that?

No, if we have a trailing slash, I think that's enough.